clang/lib/Sema/SemaChecking.cpp

```diff
@@ -1981,6 +1981,29 @@ static bool OpenCLBuiltinToAddr(Sema &S, unsigned BuiltinID, CallExpr *Call) {
   return false;
 }
 
+// In OpenCL, __builtin_alloca_* should return a pointer to address space
+// that corresponds to the stack address space i.e private address space.
+static bool OpenCLBuiltinAllocaAddrSpace(Sema &S, CallExpr *TheCall) {
+  S.Diag(TheCall->getBeginLoc(), diag::warn_alloca)
+      << TheCall->getDirectCallee();
+
```

Comment:
Or  I could place this non-OpenCL specific code just before the invocation of this function. My idea to do so was to wrap everything in a function, so just a single function call comes into picture.